modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala

```diff
@@ -0,0 +1,237 @@
+package be.wegenenverkeer.atomium.client
+
+import java.time.LocalDateTime
+
+import akka.actor.{ActorLogging, ActorRef}
+import akka.persistence.{PersistentActor, RecoveryCompleted, SnapshotOffer}
+import be.wegenenverkeer.atomium.client.ImplicitConversions._
+import be.wegenenverkeer.atomium.japi
+import be.wegenenverkeer.atomium.japi.client.FeedEntry
+import be.wegenenverkeer.atomium.client.AbstractAtomiumFeedClientActor.Protocol.HasPosition
+import com.fasterxml.jackson.databind.JsonNode
+import play.api.libs.json._
+import rx.lang.scala.{Observable, Subscriber}
+
+import scala.collection.immutable
+import scala.util.Try
+
+/**
+ * Generic Actor that consumes Atomium feeds (in json format) via the AtomiumClient.
+ * If the feed handling goes wrong, the actor will stop (whatever the failure type is).
+ * It is therefore suggested to run this actor within a BackoffSupervisor or similar construct.
+ *
+ * This actor only sends messages to itself. The feed entries are rapported by the [[publishEvent]] method.
+ *
+ * It's the job of the subclass to decide what to do with these entries.
+ *
+ * @param config feed and actor config
+ * @param eventFormat format for EventBus events
+ * @param feedEntryFormat Reads for reading entries in the feed
+ * @tparam EVENT EVENT type that will be put on the eventbus
+ * @tparam FEEDENTRY FEEDENTRY type in the feed
+ */
+abstract class AbstractAtomiumFeedClientActor[EVENT <: HasPosition, FEEDENTRY](config: FeedConfig,
+                                                                              eventFormat: Format[EVENT],
+                                                                              feedEntryFormat: Reads[FEEDENTRY],
+                                                                              monitorActor: ActorRef
+                                                                               ) extends PersistentActor with ActorLogging {
+
+  import AbstractAtomiumFeedClientActor.Protocol._
+
+  lazy val atomiumClient = new japi.client.AtomiumClient.Builder()
+    .setBaseUrl(config.baseUrl)
+    .setAcceptJson()
+    .setConnectTimeout(config.connectTimeout.toMillis.toInt)
+    .build
+    .asScala
+
+  var positionInFeed: Option[FeedPosition] = None
+
+  def receiveRecover: Receive = {
+    case RecoveryCompleted =>
+      log.debug("{}: Recovery completed, start the feed observer", config.feedUrl)
+      subscribeToFeed()
+
+    case jsValue: JsValue => onJsValueEvent(jsValue)
+
+    case x: SnapshotOffer => positionInFeed = x.snapshot.asInstanceOf[Option[FeedPosition]]
+  }
+
+  def createEventToPublish(c: NewFeedEntryReceived[FEEDENTRY]): EVENT
+
+  def publishEvent(event: EVENT): Unit
+
+  def status(atomiumFeedClientStatus: AtomiumFeedClientStatus): Unit = ()
```

Comment:
What is the point of this method? Should document wat is the point of this method.
Also, should it not be abstract?